### PR TITLE
Tests rspec

### PR DESCRIPTION
**Pull request by josuerushanika**
In this branch me and @obed300 ;

Added below unit tests:
Book
Person
Student
Teacher
Classroom
Rental
Base Decorator
Capitalize decorator
Trimmer decorator

### CODE REVIEW

File: data/spec/base_decorator_spec.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+require './nameable'
+
+describe Decorator do
+  context 'Creating decorator test methods' do
+    decorator = Decorator.new('Obed Bamfo')
+    it 'sets decorator nameable as Obed Bamfo' do
+      expect(decorator.nameable).to eql 'Obed Bamfo'
+    end
+  end
+end
```
Comment (BregornOriginal) [inline data/spec/base_decorator_spec.rb]: - [ ] Kindly check this test, it's not working

File: data/spec/capitalize_decorator_spec.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+describe capitalize_decorator do
+  context 'Capitalize Decorator' do
+    it 'capitalize the first name' do
+      person = Person.new(0o1, 'Obed')
+      capitalize_person = CapitalizeDecorator.new(person).correct_name
+      expect(capitalize_person).to eq 'Obed'
+    end
+  end
+end
```
Comment (BregornOriginal) [inline data/spec/capitalize_decorator_spec.rb]: - [ ] Kindly check this test, it's not working

File: data/spec/classroom_spec.rb [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+    it 'adds students to classroom' do
+      new_student = Student.new(24, 'Josue', true)
+      classroom.students.push(new_student)
+      expect(classroom.students).to include(new_student)
+      expect(classroom.students.length).to eql 1
+      expect(new_student.name).to eql 'Josue'
+    end
+  end
+end
```
Comment (BregornOriginal) [inline data/spec/classroom_spec.rb]: - [ ] Kindly check this test, it's not working

File: data/spec/nameable_spec.rb [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+require './person'
+
+describe Nameable do
+  context 'Testing correct name method' do
+    person = Person.new(40, 'Josue', parent_permission: false)
+    it 'should return correct name' do
+      expect(person.correct_name).to eql 'Josue'
+    end
+  end
+end
```
Comment (BregornOriginal) [inline data/spec/nameable_spec.rb]: - [ ] Kindly check this test, it's not working

File: data/spec/person_spec.rb [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+      expect(rental).to be_a Rental
+      expect(rental.date).to eql('2022-02-02')
+    end
+
+    it 'returns correct name' do
+      name = person.correct_name
+      expect(name).to eql 'Josue'
+    end
+  end
+end
```
Comment (BregornOriginal) [inline data/spec/person_spec.rb]: - [ ] Kindly check this test, it's not working

File: data/spec/rental_spec.rb [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+      expected_hash = {
+        date: '2020-02-02',
+        book: book,
+        person: person
+      }
+
+      expect(rental.to_h).to eq(expected_hash)
+    end
+  end
+end
```
Comment (BregornOriginal) [inline data/spec/rental_spec.rb]: - [ ] Kindly check this test, it's not working

File: data/spec/student_spec.rb [hunk 7]
```diff
@@ -0,0 +1,43 @@
+require './student'
+require './classroom'
+
+describe Student do
```
Comment (BregornOriginal) [inline data/spec/student_spec.rb]: - [ ] Kindly check this test, it's not working

File: data/spec/teacher_spec.rb [hunk 8]
```diff
@@ -0,0 +1,31 @@
+require './teacher'
+
+describe Teacher do
```
Comment (BregornOriginal) [inline data/spec/teacher_spec.rb]: - [ ] Kindly check this test, it's not working

File: data/spec/trimmer_decorator_spec.rb [hunk 9]
```diff
@@ -0,0 +1,14 @@
+require './nameable'
+require './person'
+require './trimmer_decorator'
+
+describe TrimmerDecorator do
```
Comment (BregornOriginal) [inline data/spec/trimmer_decorator_spec.rb]: - [ ] Kindly check this test, it's not working

**Review by BregornOriginal — CHANGES_REQUESTED** [review]
Hi @josuerushanika  

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

cat-driving-serious

## Highlights

✅ Linters are passing
✅ Good commit history
✅ README looks professional

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by tresorsawasawa — APPROVED** [review]
# STATUS: APPROVE :heavy_check_mark: :heavy_check_mark:  :champagne: 

Hi @josuerushanika and @obed300 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 :clap: :clap: :100: 

nodding-yes

## To highlight :champagne: 
- All linters are passing :heavy_check_mark: 
- All tests are now passing :heavy_check_mark: 
- Good commit history :heavy_check_mark: 
- Good PR description :heavy_check_mark: 
- Your README file is professional :heavy_check_mark: 

## Optional suggestions
_Every comment with the **[OPTIONAL]** prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please ping me [@tresorsawasawa](https://github.com/tresorsawasawa) when you comment so I can receive the notification or use slack with the same name._

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by josuerushanika** [comment]
Thank You

**Pull request merged**